Sources/Classes/SMStoreSyncOperation.swift

```diff
@@ -93,20 +93,29 @@ class SMStoreSyncOperation: Operation {
         self.operationQueue = OperationQueue()
         self.operationQueue.maxConcurrentOperationCount = 1
         self.localStoreMOC = NSManagedObjectContext(concurrencyType: NSManagedObjectContextConcurrencyType.privateQueueConcurrencyType)
-        self.localStoreMOC.persistentStoreCoordinator = self.persistentStoreCoordinator
         NotificationCenter.default.addObserver(self, selector: #selector(SMStoreSyncOperation.backingContextDidSave(notification:)), name: NSNotification.Name.NSManagedObjectContextDidSave, object: backingMOC)
+
+        self.localStoreMOC.persistentStoreCoordinator = self.persistentStoreCoordinator
         if let completionBlock = self.syncCompletionBlock {
             do {
                 try self.performSync()
                 print("Sync Performed", terminator: "\n")
+                NotificationCenter.default.removeObserver(self)
                 completionBlock(nil)
             } catch let error as NSError {
                 print("Sync Performed with Error", terminator: "\n")
+                NotificationCenter.default.removeObserver(self)
```

Comment:
You can use a `defer` clause for this

Reply:
I will just take the `removeObserver` out of the `do/catch`